extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketClient.java

```diff
@@ -101,6 +101,20 @@ public ExtensionWebSocketClient (String sourceName) {
         listener = new ExtensionWebSocketListener(this);
     }
 
+    /**
+     * Attempts to connect to the source using the given target url and authentication token.
+     *
+     * @param url   The url of the target Vantiq server.
+     * @param token The authentication token for the target namespace.
+     * @return      An {@link CompletableFuture} that completes as {@code true} when the connection to the source is
```

Comment:
Probably worth noting here how one might obtain the error that caused something to fail.